pre-compute/src/compute/pre_compute_app.rs

```diff
@@ -105,19 +121,27 @@ impl PreComputeAppTrait for PreComputeApp {
     /// This function panics if:
     /// - `pre_compute_args` is `None`.
     /// - `chain_task_id` is `None`.
-    fn download_input_files(&self) -> Result<(), ReplicateStatusCause> {
+    fn download_input_files(&self) -> Result<(), Vec<ReplicateStatusCause>> {
+        let mut exit_causes: Vec<ReplicateStatusCause> = vec![];
         let args = &self.pre_compute_args;
         let chain_task_id: &str = &self.chain_task_id;
 
-        for url in &args.input_files {
+        for (index, url) in args.input_files.iter().enumerate() {
```

Comment:
If the comment on using `filename` instead of `index` in `ReplicateStatusCause` is taken into account, this modification could be reverted